modules/ROOT/pages/syntax-quick-reference.adoc

```diff
@@ -0,0 +1,1290 @@
+----
+include::macros:example$url.adoc[tag=b-windows]
+----
+
+.Inline anchors
+----
+include::attributes:example$id.adoc[tag=anchor]
+----
+
+.Cross references
+[#ex-xrefs]
+----
+include::macros:example$xref.adoc[tag=b-base]
+----
+
+.View result of <<ex-xrefs>>
+[%collapsible.result]
+====
+include::macros:example$xref.adoc[tag=b-base]
+====
+
+.Inter-document cross references
+----
+include::macros:example$xref.adoc[tag=b-inter]
+----
+
+== Images
+
+Images are resolved relative to the value of the xref:macros:images-directory.adoc[imagesdir] document attribute, which is empty by default.
+Use the `imagesdir` attribute to avoid hard coding the common path to your images in every image macro.
+The `imagesdir` attribute can be an absolute path, relative path, or base URL.
+When the image target is a URL or absolute path, the imagesdir prefix is _not_ prepended.
+
+.Block image macro
+[#ex-image-blocks]
+----
+include::macros:example$image.adoc[tag=base]
+
+include::macros:example$image.adoc[tag=alt]
+
+include::macros:example$image.adoc[tag=attr]
+
+include::macros:example$image.adoc[tag=ab-url]
+----
+
+.View result of <<ex-image-blocks>>
+[%collapsible.result]
+====
+include::macros:example$image.adoc[tag=qr-base]
+
+include::macros:example$image.adoc[tag=qr-alt]
+
+include::macros:example$image.adoc[tag=qr-attr]
+
+include::macros:example$image.adoc[tag=ab-url]
+====
+
+Two colons following the image keyword in the macro (i.e., `image::`) indicates a block image (aka figure), whereas one colon following the image keyword (i.e., `image:`) indicates an inline image.
+(All macros follow this pattern).
+You use an inline image when you need to place the image in a line of text.
+Otherwise, you should prefer the block form.
+
+.Inline image macro
+[#ex-image-inline]
+----
+include::macros:example$image.adoc[tag=inline]
+----
+
+.View result of <<ex-image-inline>>
+[%collapsible.result]
+====
+include::macros:example$image.adoc[tag=qr-inline]
+====
+
+.Inline image macro with positioning role
+[#ex-image-role]
+----
+include::macros:example$image.adoc[tag=in-role]
+----
+
+.View result of <<ex-image-role>>
+[%collapsible.result]
+====
+include::macros:example$image.adoc[tag=qr-role]
+====
+
+.Embedded
+----
+include::macros:example$image.adoc[tag=data]
+----
+
+When the `data-uri` attribute is set, all images in the document--including admonition icons--are embedded into the document as {url-data-uri}[data URIs].
+You can also pass it as a command line argument using `-a data-uri`.
+
+== Videos
+
+.Block video macro
+----
+include::macros:example$video.adoc[tag=base]
+
+include::macros:example$video.adoc[tag=attr]
+----
+
+.Embedded Youtube video
+----
+include::macros:example$video.adoc[tag=you]
+----
+
+.Embedded Vimeo video
+----
+include::macros:example$video.adoc[tag=vim]
+----
+
+You can control the video settings using xref:macros:audio-and-video.adoc[additional attributes and options] on the macro.
+
+== Source Code
```

Comment:
The title is a bit misleading since we are also showing a literal block that contains output from a program:

```
....
error: 1954 Forbidden search
absolutely fatal: operation lost in the dodecahedron of doom

Would you like to try again? y/n
....
```

Also, using monospaced is not really "source code".
Maybe, Monospaced Text and Terms? Technical Text and Terms? Something like that... :thinking:

Reply:
@mojavelinux Ooops, just saw your previous comment:

> The "Source Code" section seems to be misleading.
> ...
> Verbatim
> Verbatim Content and Source Code
> Literals and Source Code

I like the word "Verbatim" but we don't use it often. So it could create confusion Verbatim vs Literals. I also think that a lot of people will search for "source code" so I'd vote for "Literals and Source Code".

Reply:
> I like the word "Verbatim" but we don't use it often. So it could create confusion Verbatim vs Literals. I also think that a lot of people will search for "source code" so I'd vote for "Literals and Source Code".

I also think a lot of people search/skim for "source code" so it should be included in the title. "Literals and Source Code" seems clear and to the point that people would have a good idea what the topic includes/what they would find under it. My next question is, should we tweak the name of the navigation entry in the AsciiDoc Docs from "Verbatim and Source Blocks" to something like "Literal and Source Blocks" or "Literal and Source Code Blocks"?

Reply:
I think we should. A listing block is a literal block at a technical level, just one that gets some extra behavior (namely a caption). So "Literal and Source Code Blocks" or even "Literal and Code Blocks" seems reasonable. For reference, CommonMark calls them all "Code Blocks".

We're going to trip over some words trying to find the right combination, but we'll get it.